src/workflows-model.ts

```diff
@@ -192,6 +191,49 @@ export interface JobDefaults {
   readonly run?: RunSettings;
 }
 
+/**
+ * Enum to indicate the runner type
+ */
+export enum RunnerType {
+  GITHUB_HOSTED,
+  SELF_HOSTED,
+}
+
+/**
+ * The type of runner to run the job on. Can be GitHub or Self-hosted.
+ * In case of self-hosted, a list of labels can be supplied.
+ */
+export class Runner {
+  public static readonly UBUNTU_LATEST = new Runner(
+    'ubuntu-latest',
+    RunnerType.GITHUB_HOSTED,
+  );
+
+  public static readonly WINDOWS_LATEST = new Runner(
+    'windows-latest',
+    RunnerType.GITHUB_HOSTED,
+  );
+
+  public static readonly MACOS_LATEST = new Runner(
+    'macos-latest',
+    RunnerType.GITHUB_HOSTED,
+  );
+
+  public static selfHosted(labels: string[]): Runner {
+    return new Runner('self-hosted', RunnerType.SELF_HOSTED, labels);
+  }
+
+  public readonly runsOn: string;
+  public readonly runnerType: RunnerType;
+  public readonly labels: string[];
+
+  constructor(runsOn: string, runnerType: RunnerType, labels?: string[]) {
```

Comment:
Let's make the constructor private and get rid of `RunnerType`

Reply:
Done